### ISSUE TITLE
Attach returns system box instead of system

### ISSUE THREAD

**Issue by Myrannas** [LastOliveGames/becsy#4]
It looks like when using the `System.attach` method, the returned value is actually the system box containing the system.

system.ts:218
```
        CHECK: if (!targetSystem) {
          throw new Error(`Attached system ${targetSystemType.name} not defined in this world`);
        }
        (this.system as any)[prop] = targetSystem;
      }
```

I think it should be `targetSystem.system` instead.

I have the code (and associated test fixed) on a fork, do you mind me raising a PR?

**Comment by pkaminski** [comment]
I have that fix pending and will be cutting a new release very shortly.  :)

**Comment by pkaminski** [comment]
Fixed in 0.8.2.

**Comment by Myrannas** [comment]
Oh, haha, thanks :D!
